checks/raw/shell_download_validate.go

```diff
@@ -418,50 +416,39 @@ func isNpmUnpinnedDownload(cmd []string) bool {
 	return false
 }
 
-func isGoUnpinnedDownload(cmd []string) bool {
-	if len(cmd) == 0 {
-		return false
-	}
-
-	if !isBinaryName("go", cmd[0]) {
-		return false
-	}
+func isGoDownload(cmd []string) bool {
 	// `Go install` will automatically look up the
 	// go.mod and go.sum, so we don't flag it.
 	if len(cmd) <= 2 {
 		return false
 	}
 
-	found := false
+	return isBinaryName("go", cmd[0]) && slices.Contains([]string{"get", "install"}, cmd[1])
+}
+
+func isGoUnpinnedDownload(cmd []string) bool {
 	insecure := false
 	hashRegex := regexp.MustCompile("^[A-Fa-f0-9]{40,}$")
 	semverRegex := regexp.MustCompile(`^v\d+\.\d+\.\d+(-[0-9A-Za-z-.]+)?(\+[0-9A-Za-z-.]+)?$`)
-	for i := 1; i < len(cmd)-1; i++ {
-		// Search for get and install commands.
-		if slices.Contains([]string{"get", "install"}, cmd[i]) {
-			found = true
-		}
 
-		if !found {
-			continue
+	for i := 2; i < len(cmd); i++ {
+		// The -d flag instructs get to download but not install the packages
+		if strings.EqualFold(cmd[i], "-d") {
```

Comment:
Why `EqualFold` over `Equal`? Not sure that `-D` is a valid flag.

Reply:
About using `EqualFold`, there's no special reason for this choice, just copied from another code snippet. I did not find the method `strings.Equal` in https://pkg.go.dev/strings tho. Am I getting smth wrong?

About `-d` flag, I got that info from this doc: https://pkg.go.dev/cmd/go/internal/get . As far as I understand `-d` is a valid flag when doing `go get`. And this flag does only the download, not installation of packages. As I understand, the installation is the one considered possibly harmful, so using `-d` flag is considered safe.